Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp

```diff
@@ -171,11 +171,13 @@ void WebFullScreenManager::enterFullScreenForElement(WebCore::Element* element)
 
     setElement(*element);
 
-    bool isVideoElement = false;
+    bool isVideoElementWithControls = false;
 #if PLATFORM(IOS_FAMILY) || (PLATFORM(MAC) && ENABLE(VIDEO_PRESENTATION_MODE))
+    if (auto* videoElement = dynamicDowncast<HTMLVideoElement>(element))
+        isVideoElementWithControls = videoElement->controls();
```

Comment:
should we have this be an `else` in case we do have a `currentPlaybackControlsElement` (or if it doesn't match the `element`)?

Reply:
> should we have this be an else in case we do have a currentPlaybackControlsElement (or if it doesn't match the element)?

I don't think so, the intent is to hide the controls on the UIProcess-side, if the element being full screened (`element`) already has controls. I believe the logic I have captures that case.

The duplicated controls are specific to `video` elements, so we do not need to handle `audio`.